### PR TITLE
Update docker 1.10

### PR DESCRIPTION
**Pull request by gregarndt**

### CODE REVIEW

**Comment by gregarndt** [comment]
Quick graph executing 1000 tasks: https://tools.taskcluster.net/task-graph-inspector/#lXuFfeHjRjWPvPBPKGs2Qg

**Comment by djmitche** [comment]
This is working great for us.  Please merge! :)

File: lib/task.js [hunk 1]
```diff
@@ -333,7 +333,8 @@ export class Task {
         StdinOnce: false,
         Env: taskEnvToDockerEnv(env),
         HostConfig: {
-          Privileged: privilegedTask
+          Privileged: privilegedTask,
+          ShmSize: 1800000000
```
Comment (djmitche) [inline lib/task.js]: Or, if you want to make this a task argument before merging, that would be cool too.

File: deploy/packer/base/scripts/packages.sh [hunk 2]
```diff
@@ -1,6 +1,11 @@
 #! /bin/bash
 
-set -e -x
+set -e -v
+
+DOCKER_VERSION=1.10.1-0~trusty
+# Kernels <= 3.13.0.77 and >= 3.13.0.71 have an AUFS bug which can cause docker
+# containers to not exit properly because of zombie processes that can't be reaped.
+KERNEL_VERSION=3.13.0-79-generic
```
Comment (djmitche) [inline deploy/packer/base/scripts/packages.sh]: I'm impressed -- where'd you find this?

Comment (gregarndt) [inline deploy/packer/base/scripts/packages.sh]: Found while reading this issue: https://github.com/docker/docker/issues/18180

Comment (gregarndt) [inline deploy/packer/base/scripts/packages.sh]: credit to the folks in #docker on freenode

File: deploy/packer/base/scripts/packages.sh [hunk 3]
```diff
@@ -14,24 +20,58 @@ sudo apt-get update -y
...
-    lxc-docker-1.6.1 \
+    docker-engine=$DOCKER_VERSION \
     btrfs-tools \
     lvm2 \
     curl \
     build-essential \
-    linux-image-extra-`uname -r` \
-    linux-image-extra-virtual \
     git-core \
+    gstreamer0.10-alsa \
```
Comment (gregarndt) [inline deploy/packer/base/scripts/packages.sh]: This is useful for debugging video loopback issues on the host.

File: deploy/packer/base/scripts/packages.sh [hunk 4]
```diff
@@ -41,8 +81,30 @@ sudo apt-get install -y \
...
     rsyslog-gnutls \
     openvpn \
-    v4l2loopback-utils \
     lxc
 
 ## Clear mounts created in base image so fstab is empty in other builds...
 sudo sh -c 'echo "" > /etc/fstab'
+
+
+## Install v4l2loopback
```
Comment (gregarndt) [inline deploy/packer/base/scripts/packages.sh]: Moved out some of the setup that was done when building the app ami to here so it's clear what is being configured.

File: vagrant.sh [hunk 5]
```diff
@@ -1,30 +1,75 @@
-#! /bin/bash -vex
+#! /bin/bash
```
Comment (gregarndt) [inline vagrant.sh]: Most of this is to make it match the ami setup a little more

**Comment by gregarndt** [comment]
so I tested out ptrace manually on some hosts and it appears to work as expected.  It's the last test failing on docker 1.10 and I can run the tests correctly in vagrant, but not within the worker-ci docker-in-docker image.  It looks like apprmor stepping on itself and even messaged up apparmor on the host.

I need to look into using the newer dind images that docker-library has and see if we can use that for the CI tests.

**Comment by gregarndt** [comment]
@djmitche Mind giving this a look over

File: deploy/packer/base/scripts/packages.sh [hunk 6]
```diff
@@ -14,24 +20,58 @@ sudo apt-get update -y
...
+timeout         0
+hiddenmenu
+
+title           Ubuntu 14.04.2 LTS, kernel ${KERNEL_VER}
+root            (hd0)
+kernel          /boot/vmlinuz-${KERNEL_VER} root=LABEL=cloudimg-rootfs ro console=hvc0
+initrd          /boot/initrd.img-${KERNEL_VER}
+
+
+## Update to pick up new registries
```
Comment (djmitche) [inline deploy/packer/base/scripts/packages.sh]: "registries"?

File: deploy/packer/base/scripts/packages.sh [hunk 7]
```diff
@@ -41,8 +81,30 @@ sudo apt-get install -y \
...
+
+cat <<EOF | sudo tee /etc/modprobe.d/test-modules.conf >&2
+options v4l2loopback devices=100
+EOF
+
+
+# Install Audio loopback devices
+echo "snd-aloop" | sudo tee --append /etc/modules
+
+cat <<EOF | sudo tee /etc/modprobe.d/test-modules.conf >&2
```
Comment (djmitche) [inline deploy/packer/base/scripts/packages.sh]: `tee --append`?

**Comment by djmitche** [comment]
looks good

File: deploy/packer/base/scripts/packages.sh (hunk 6 above)
Comment (gregarndt) [inline deploy/packer/base/scripts/packages.sh]: Ah, this was some bad merge work I was doing.  I'll remove that.

**Pull request merged**